File: mobile/iosApp/Bissbilanz/Models/ParsedNutrition.swift
import Foundation

/// Nutrition values extracted from an OCR'd nutrition-facts panel, normalized
/// to a per-100 g / per-100 ml basis. Every field is optional — OCR is
/// best-effort, so the user always confirms and edits the result in
/// `FoodEditSheet` before it is saved.
///
/// Units follow the `FoodCreate` convention: macros and `salt` in grams,
/// `sodium` in milligrams.
struct ParsedNutrition: Equatable {
    var calories: Double?
    var protein: Double?
    var carbs: Double?
    var fat: Double?
    var fiber: Double?
    var sugar: Double?
    var saturatedFat: Double?
    var salt: Double?
    var sodium: Double?

    /// True when nothing usable was parsed — the scan surfaces an error
    /// instead of opening an empty confirmation sheet.
    var isEmpty: Bool {
        calories == nil
            && protein == nil
            && carbs == nil
            && fat == nil
            && fiber == nil
            && sugar == nil
            && saturatedFat == nil
            && salt == nil
            && sodium == nil
    }

    /// True when at least one of the headline values (calories + the four
    /// macros) was found — used to decide whether the iOS 26 document path
    /// produced a good-enough result or should fall back to the line path.
    var hasCoreMacros: Bool {
        calories != nil || protein != nil || carbs != nil || fat != nil
    }
}

File: mobile/shared/src/commonMain/kotlin/com/bissbilanz/label/ParsedNutrition.kt
package com.bissbilanz.label

/**
 * Nutrition values extracted from an OCR'd nutrition-facts panel, normalized to
 * a per-100 g / per-100 ml basis. Every field is optional — OCR is best-effort,
 * so the user always confirms and edits the result in the food edit sheet
 * before it is saved.
 *
 * Units follow the `FoodCreate` convention: macros and `salt` in grams,
 * `sodium` in milligrams. Mirrors the iOS `ParsedNutrition`.
 */
data class ParsedNutrition(
    var calories: Double? = null,
    var protein: Double? = null,
    var carbs: Double? = null,
    var fat: Double? = null,
    var fiber: Double? = null,
    var sugar: Double? = null,
    var saturatedFat: Double? = null,
    var salt: Double? = null,
    var sodium: Double? = null,
) {
    /**
     * True when nothing usable was parsed — the scan surfaces an error instead
     * of opening an empty confirmation sheet.
     */
    val isEmpty: Boolean
        get() =
            calories == null &&
                protein == null &&
                carbs == null &&
                fat == null &&
                fiber == null &&
                sugar == null &&
                saturatedFat == null &&
                salt == null &&
                sodium == null

    /**
     * True when at least one of the headline values (calories + the four macros)
     * was found.
     */
    val hasCoreMacros: Boolean
        get() = calories != null || protein != null || carbs != null || fat != null
}
